dll/win32/uxtheme/nonclient.c

```diff
@@ -1121,22 +1239,88 @@ HRESULT WINAPI DrawNCPreview(HDC hDC,
     context.scrolltheme = OpenThemeDataFromFile(hThemeFile, hwndDummy, L"SCROLLBAR", 0);
     if (!context.scrolltheme)
         return E_FAIL;
-    context.Active = TRUE;
     context.wi.cbSize = sizeof(context.wi);
     if (!GetWindowInfo(hwndDummy, &context.wi))
         return E_FAIL;
     context.wi.dwStyle |= WS_VISIBLE;
-    context.CaptionHeight = context.wi.cyWindowBorders;
-    context.CaptionHeight += GetSystemMetrics(context.wi.dwExStyle & WS_EX_TOOLWINDOW ? SM_CYSMCAPTION : SM_CYCAPTION );
+
     context.hRgn = CreateRectRgnIndirect(&context.wi.rcWindow);
+    RECT rcAdjPreview = { prcPreview->left, prcPreview->top, prcPreview->right, prcPreview->bottom };
+    INT previewWidth = rcAdjPreview.right - rcAdjPreview.left;
+    INT previewHeight = rcAdjPreview.bottom - rcAdjPreview.top;
 
-    /* Paint the window on the preview hDC */
-    rcCurrent = context.wi.rcWindow;
-    OffsetRect( &rcCurrent, -context.wi.rcWindow.left, -context.wi.rcWindow.top);
-    SetViewportOrgEx(hDC, context.wi.rcWindow.left, context.wi.rcWindow.top, NULL);
-    ThemeCalculateCaptionButtonsPos(hwndDummy, context.theme);
-    ThemePaintWindow(&context, &rcCurrent, FALSE);
-    SetViewportOrgEx(hDC, 0, 0, NULL);
+    /* Draw inactive preview window */
+    context.Active = FALSE;
+    SetWindowTextW(hwndDummy, L"Inactive Window");
+    DrawWindowForNCPreview(hDC, &context, rcAdjPreview.left, rcAdjPreview.top, rcAdjPreview.right - 17, rcAdjPreview.bottom - 20, TRUE, NULL);
```

Comment:
These magic numbers (and in the other instances of DrawWindowForNCPreview), do they come from themes, was it something that appears nice visually, or .... ?

They should probably be scaled depending on dpi, or be derived from some metric?

Reply:
They're not supposed to be derived from the visual style or metrics, no. As for DPI scaling...maybe? But we don't exactly have DPI scaling support yet, so...is there even anything to do on that front for now?

Reply:
I wish we could cross-test uxtheme.dll from ReactOS in Windows... 🤤

Reply:
@binarymaster Yeah, me too...and believe me, I've tried.

Getting our `uxtheme.dll` to load *at all* was actually hilariously easy on Vista, but took a lot more work (and some much-needed help from @DarkFire01) to make happen on Server 2003. Getting it to actually *apply visual styles*, on the other hand...no dice so far.

Reply:
I had some limited success with shell component testing by loading the windows dll next to the reactos dll, and redirecting individual functions from one to the other (using the library `Minhook`.
That is completely out of scope for this PR though :)